source/client.c

```diff
@@ -289,10 +311,13 @@ struct aws_mqtt_client_connection *aws_mqtt_client_connection_new(struct aws_mqt
 
     assert(client);
 
+    AWS_LOGF_DEBUG(AWS_LS_MQTT_CLIENT, "Creating new connection");
+
     struct aws_mqtt_client_connection *connection =
         aws_mem_acquire(client->allocator, sizeof(struct aws_mqtt_client_connection));
 
     if (!connection) {
+        AWS_LOGF_ERROR(AWS_LS_MQTT_CLIENT, "Failed to allocate connection object");
```

Comment:
Usually don't bother with logging allocation failures since an allocation failure means the log command itself will most likely fail

Reply:
That makes sense